code/game/objects/structures/stool_bed_chair_nest/sofa.dm

```diff
@@ -0,0 +1,208 @@
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_BEIGE)
+
+/obj/structure/bed/stool/chair/sofa/left/red/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CARPET)
+
+/obj/structure/bed/stool/chair/sofa/left/orange/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_ORANGE)
+
+/obj/structure/bed/stool/chair/sofa/left/yellow/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_YELLOW)
+
+/obj/structure/bed/stool/chair/sofa/left/green/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_GREEN)
+
+/obj/structure/bed/stool/chair/sofa/left/lime/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_LIME)
+
+/obj/structure/bed/stool/chair/sofa/left/blue/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_BLUE)
+
+/obj/structure/bed/stool/chair/sofa/left/teal/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_TEAL)
+
+/obj/structure/bed/stool/chair/sofa/left/purple/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_PURPLE)
+
+/obj/structure/bed/stool/chair/sofa/left/violet/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_VIOLET)
+
+/obj/structure/bed/stool/chair/sofa/right
+	icon_state = "sofa_end_right"
+	base_icon = "sofa_end_right"
+
+/obj/structure/bed/stool/chair/sofa/right/brown/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_LEATHER)
+
+/obj/structure/bed/stool/chair/sofa/right/black/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_DARK_GRAY)
+
+/obj/structure/bed/stool/chair/sofa/right/beige/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_BEIGE)
+
+/obj/structure/bed/stool/chair/sofa/right/red/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CARPET)
+
+/obj/structure/bed/stool/chair/sofa/right/orange/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_ORANGE)
+
+/obj/structure/bed/stool/chair/sofa/right/yellow/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_YELLOW)
+
+/obj/structure/bed/stool/chair/sofa/right/green/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_GREEN)
+
+/obj/structure/bed/stool/chair/sofa/right/lime/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_LIME)
+
+/obj/structure/bed/stool/chair/sofa/right/blue/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_BLUE)
+
+/obj/structure/bed/stool/chair/sofa/right/teal/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_TEAL)
+
+/obj/structure/bed/stool/chair/sofa/right/purple/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_PURPLE)
+
+/obj/structure/bed/stool/chair/sofa/right/violet/New(var/newloc)
+	..(newloc, MATERIAL_STEEL, MATERIAL_CLOTH, COLOR_VIOLET)
+
+
+/obj/structure/bed/stool/chair/sofa/corner
+	name = "impossible sofa corner"
+	desc = "This kind of sofa shouldn't even exist at all. If you see this non-euclidean specimen, contact your station's Anti-Couch Surfing Department."
```

Comment:
This should probably be a purely OOC description to report a bug.

Reply:
This thing shouldn't be obtainable in game in the first place anyway